src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -145,6 +165,51 @@ export class AccessibleBufferWidget extends DisposableStore {
 		this._bufferEditor.setScrollTop(this._bufferEditor.getScrollHeight());
 	}
 
+	async showCommandQuickPick(): Promise<void> {
+		if (!this._focusedContextKey.get()) {
+			await this.show();
+		}
+		this._inQuickPick = true;
+		const commands = this._instance.capabilities.get(TerminalCapability.CommandDetection)?.commands;
+		const quickPick = this._quickInputService.createQuickPick<IQuickPickItem>();
+		const quickPickItems: IQuickPickItem[] = [];
+		if (!commands?.length) {
+			return;
+		}
+		for (const command of commands) {
+			const line = command.marker?.line;
+			if (!line) {
+				continue;
+			}
+			quickPickItems.push(
+				{
+					label: localize('terminal.integrated.symbolQuickPick.labelNoExitCode', '{0}', command.command),
+					meta: JSON.stringify({ line: line + 1, exitCode: command.exitCode })
+				});
+		}
+		quickPick.onDidAccept(() => {
+			const item = quickPick.activeItems[0];
+			const model = this._bufferEditor.getModel();
+			if (!model || !item.meta) {
+				return;
+			}
+			quickPick.hide();
+			const data: { line: number; exitCode: number } = JSON.parse(item.meta);
+			this._bufferEditor.setSelection({ startLineNumber: data.line, startColumn: 1, endLineNumber: data.line, endColumn: 1 });
+			this._bufferEditor.revealLine(data.line);
+			this._inQuickPick = false;
+			return;
+		});
+		quickPick.onDidChangeActive(() => {
+			const data = quickPick.activeItems?.[0]?.meta;
+			if (data && JSON.parse(data).exitCode) {
+				this._audioCueService.playAudioCue(AudioCue.error, true);
+			}
+		});
+		quickPick.items = quickPickItems.reverse();
+		quickPick.show();
```

Comment:
Little thing, we normally resolve the command promise when the action is complete, like this: https://github.com/microsoft/vscode/blob/6c771e3377125c4fa7097c6a5a3d885e38dc4f88/src/vs/workbench/contrib/terminal/browser/terminalActions.ts#L402-L407

Doing this would allow an extension to chain together commands, it's would also tie the quick pick's lifecycle to the command resolving which feels nicer than it being detached from that.